cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -616,6 +619,276 @@ async def _get_trustworthiness_score_async(
+                This method will raise an exception if any errors occur or if you hit a timeout (given a timeout is specified).
+                Use it if you want strict error handling and immediate notification of any exceptions/timeouts.
+
+                If running this method on a big batch of prompts: you might lose partially completed results if TLM fails on any one of them.
+                To avoid losing partial results for the prompts that TLM did not fail on,
+                you can either call this method on smaller batches of prompts at a time
+                (and save intermediate results between batches), or use the [`try_prompt()`](#method-try_prompt) method instead.
+        """
+        prompt_response = self.tlm_response.prompt(prompt)
+
+        if isinstance(prompt_response, list):
+            assert isinstance(prompt, Sequence)
+            response = [r["response"] for r in prompt_response]
+            perplexity = [r["log"]["perplexity"] for r in prompt_response]
+            return self._batch_score(prompt, response, perplexity)
+
+        assert isinstance(prompt, str)
+        return self._score(
+            prompt,
+            prompt_response["response"],
+            perplexity=prompt_response["log"]["perplexity"],
+        )
+
+    def try_prompt(
+        self,
+        prompt: Sequence[str],
+    ) -> List[Optional[TLMResponse]]:
+        """
+        Gets response and trustworthiness score for any batch of prompts,
+        handling any failures (errors of timeouts) by returning None in place of the failures.
+
+        The list returned will have the same length as the input list, if there are any
+        failures (errors or timeout) processing some inputs, the list will contain None in place of the response.
+
+        This is the recommended way to get TLM responses and trustworthiness scores for big datasets of many prompts,
+        where some individual TLM responses within the dataset may fail. It ensures partial results are not lost.
+
+        Args:
+            prompt (Sequence[str]): list of multiple prompts for the TLM
+        Returns:
+            List[Optional[TLMResponse]]: list of [TLMResponse](#class-tlmresponse) objects containing the response and trustworthiness score.
+                The returned list will always have the same length as the input list.
+                In case of TLM failure on any prompt (due to timeouts or other errors),
+                the return list will contain None in place of the TLM response for that failed prompt.
+                Use this to obtain TLM results for as many prompts as possible,
+                but you might miss out on certain error messages.
+                If you prefer to be notified immediately about any errors or timeouts when running many prompts,
+                use the [`prompt()`](#method-prompt) method instead.
+        """
```

Comment:
Instead of all this could we just point the reader to the TLM prompt() documentation?